src/test/java/teammates/test/cases/common/FieldValidatorTest.java

```diff
@@ -481,77 +494,73 @@ public void TestGetValidityInfo_INSTRUCTOR_ROLE() {
-                FieldType.EMAIL, 
-                "e@y", 
-                "");
-        
-        String maxLengthValue = StringHelper.generateStringOfLength(EMAIL_MAX_LENGTH - 6) + "@c.gov";
-        testOnce("valid: max length", 
-                FieldType.EMAIL, 
-                maxLengthValue, 
-                "");
+    public void testGetValidityInfoEmail() {
+        invalidityInfoFor_nullEmail_shouldThrowException();
+        invalidityInfoFor_validEmail_shouldReturnEmptyString();
+        invalidityInfoFor_invalidEmail_shouldReturnErrorString();
+    }
 
-        String emptyValue = "";
-        testOnce("invalid: empty string", 
-                FieldType.EMAIL, 
-                emptyValue, 
-                String.format(EMAIL_ERROR_MESSAGE, emptyValue,    REASON_EMPTY));
-        
-        String untrimmedValue = " e@email.com ";
-        testOnce("invalid: untrimmed", 
-                FieldType.EMAIL, 
-                untrimmedValue,
-                String.format(WHITESPACE_ONLY_OR_EXTRA_WHITESPACE_ERROR_MESSAGE, "email"));
-        
-        String whitespaceOnlyValue = "    ";
-        testOnce("invalid: whitespace only", 
-                FieldType.EMAIL, 
-                whitespaceOnlyValue,
-                String.format(WHITESPACE_ONLY_OR_EXTRA_WHITESPACE_ERROR_MESSAGE, "email"));
-        
-        String tooLongValue = maxLengthValue + "x";
-        testOnce("invalid: too long", 
-                FieldType.EMAIL, 
-                tooLongValue, 
-                String.format(EMAIL_ERROR_MESSAGE, tooLongValue, REASON_TOO_LONG));
-        
-        String valueWithDisallowedChar = "woMAN@com. sg";
-        testOnce("invalid: disallowed char (space) after @", 
-                FieldType.EMAIL, 
-                valueWithDisallowedChar, 
-                String.format(EMAIL_ERROR_MESSAGE, valueWithDisallowedChar, REASON_INCORRECT_FORMAT));
-        
-        valueWithDisallowedChar = "man woman@com.sg";
-        testOnce("invalid: disallowed char (space)", 
-                FieldType.EMAIL, 
-                valueWithDisallowedChar, 
-                String.format(EMAIL_ERROR_MESSAGE, valueWithDisallowedChar, REASON_INCORRECT_FORMAT));
-        
-        valueWithDisallowedChar = "man@woman@com.lk";
-        testOnce("invalid: multiple '@' signs", 
-                FieldType.EMAIL, 
-                valueWithDisallowedChar, 
-                String.format(EMAIL_ERROR_MESSAGE, valueWithDisallowedChar, REASON_INCORRECT_FORMAT));
-        
+    private void invalidityInfoFor_nullEmail_shouldThrowException() {
+        String errorMessage = "Did not throw the expected AssertionError for null email";
+        try {
+            validator.getInvalidityInfoForEmail(null);
+            signalFailureToDetectException(errorMessage);
+        } catch (AssertionError e) {
+            ignoreExpectedException();
+        }
     }
-    
+
+    private void invalidityInfoFor_validEmail_shouldReturnEmptyString() {
```

Comment:
I think the word `should` is redundant because the 3rd part of the name is 'expected result'.